archivers/lzlib/Portfile

```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+
+name                lzlib
+version             1.14
+revision            0
+
+homepage            https://www.nongnu.org/lzip/lzlib.html
+
+categories          archivers devel
+installs_libs       yes
+license             BSD
+maintainers         {gmail.com:herby.gillot @herbygillot} \
+                    openmaintainer
+
+description         \
+    Lzlib is a data compression library providing in-memory LZMA compression \
+    and decompression functions
+
+long_description        \
+    {*}${description}, including integrity checking of the decompressed data. \
+    The compressed data format used by the library is the lzip format. Lzlib \
+    is written in C. The lzip file format is designed for data sharing and \
+    long-term archiving, taking into account both data integrity and decoder \
+    availability\: The lzip format provides very safe integrity checking and \
+    some data recovery means. The program lziprecover can repair bit flip \
+    errors \(one of the most common forms of data corruption\) in lzip files, \
+    and provides data recovery capabilities, including error-checked merging \
+    of damaged copies of a file. The lzip format is as simple as possible \
+    \(but not simpler\). The lzip manual provides the source code of a simple \
+    decompressor along with a detailed explanation of how it works, so that \
+    with the only help of the lzip manual it would be possible for a digital \
+    archaeologist to extract the data from a lzip file long after quantum \
+    computers eventually render LZMA obsolete. Additionally the lzip \
+    reference implementation is copylefted, which guarantees that it will \
+    remain free forever.
+
+checksums           rmd160  aab868b92c87268ace133d0db2789cb5ab8de8e0 \
+                    sha256  5acac8714ed4f306020bae660dddce706e5f8a795863679037da9fe6bf4dcf6f \
+                    size    131624
+
+master_sites        nongnu:lzip/lzlib
+
+platform darwin {
+    patchfiles-append patch-darwin-Makefile.in.diff
+
+    post-build {
+        set libname liblz.${version}.dylib
+        set libpath ${prefix}/lib/${libname}
+        system "install_name_tool -id ${libpath} ${worksrcpath}/${libname}"
+    }
```

Comment:
It looks like the build system didn't have any support for macOS dylibs. Please file an upstream bug report or pull request and include the URL in the portfile or patch.

You're already patching the build so it would be better to patch it more to use the correct install name to begin with (with the `-install_name` flag) rather than fix it later with `install_name_tool`. `-current_version` and `-compatibility_version` should also be specified if the project defines such values but since they don't have macOS build support they probably don't.

Using the complete project version number in the install name is usually wrong. In the Makefile it looks like there are two separate variables, `pkgversion` and `soversion`. `pkgversion` is the full project version (1.14), while `soversion` (1) is what is used by the `--soname` flag, which is equivalent to the install name on macOS.

Reply:
> Using the complete project version number in the install name is usually wrong. In the Makefile it looks like there are two separate variables, pkgversion and soversion. pkgversion is the full project version (1.14), while soversion (1) is what is used by the --soname flag, which is equivalent to the install name on macOS.

This has been addressed; now using `soversion` with `-install_name` now.